### PR TITLE
integration test for secrets management

### PR DESCRIPTION
**Pull request by protatremy**
## Contributor Checklist:

* [x] I have updated the unit tests
* [ ] I have created a file in the master/buildbot/newsfragment directory (and read the README.txt in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

**Comment by tardyp** [comment]
The smoke tests are broken. error suggest that you forgot to include your new package in the setup.py list of packages (i.e. buildbot's list of sub modules)

File: master/buildbot/config.py [hunk 1]
```diff
@@ -586,6 +589,15 @@ def load_metrics(self, filename, config_dict):
             else:
                 self.metrics = metrics
 
+    def load_secrets(self, filename, config_dict):
+        # we don't try to validate metrics keys
```
Comment (tardyp) [inline master/buildbot/config.py]: I think this is a copy pasted comment

File: master/buildbot/test/util/integration.py [hunk 2]
```diff
@@ -149,6 +149,9 @@ def dump():
                 raise self.failureException(dump.getvalue())
         self.addCleanup(dump)
 
+    def reconfigMaster(self, config_dict):
+        self.master.reconfig()
```
Comment (tardyp) [inline master/buildbot/test/util/integration.py]: I dont think this method is useful nor implemented correctly

File: master/buildbot/test/integration/test_integration_secrets.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        yield self.setupConfig(masterConfig())
+        build = yield self.doForceBuild(wantSteps=True, wantLogs=True)
+        self.assertEqual(build['buildid'], 1)
+        res = yield self.checkBuildStepLogExist(build, "echo bar")
+        self.assertTrue(res)
+
+    @defer.inlineCallbacks
+    def test_secretReconfig(self):
+        yield self.setupConfig(masterConfig())
+        self.reconfigMaster(masterConfig())
```
Comment (tardyp) [inline master/buildbot/test/integration/test_integration_secrets.py]: following is enough:

    yield self.master.reconfig()

but you mush modify the config to change the secret or this test does not test much

Comment (tardyp) [inline master/buildbot/test/integration/test_integration_secrets.py]: c = masterConfig()
     yield self.setupConfig(c)
     c['secretsProviders'] = [FakeSecretStorage(
         {"foo": "different_value", "something": "more"})]
     yield self.master.reconfig()
     [...]
     res = yield self.checkBuildStepLogExist(build, "echo different_value")

File: master/buildbot/config.py (hunk 1 above)
Comment (protatremy) [inline master/buildbot/config.py]: oups...

File: master/buildbot/config.py [hunk 4]
```diff
@@ -586,6 +589,14 @@ def load_metrics(self, filename, config_dict):
             else:
                 self.metrics = metrics
 
+    def load_secrets(self, filename, config_dict):
+        if 'secretsProviders' in config_dict:
+            secretsProviders = config_dict["secretsProviders"]
+            if not isinstance(secretsProviders, list):
+                error("c['secretsProviders'] must be a dictionary")
```
Comment (tardyp) [inline master/buildbot/config.py]: must be a list!

**Pull request merged**